WordPress/gutenberg
# Improve Inserter block hover performance

## Description
When moving our mouse over the blocks in the inserter it doesn't feel smooth. These changes aim to make it a smoother experience by avoiding unnecessary rerenders.

## How has this been tested?
* `yarn dev`
* Open Post or Site editor
* Open Inserter
* Keep moving your mouse between blocks - the preview should appear faster and it should be a better experience

## Screenshots
Profiling and GIFs were done in the Site Editor. Both the performance gains are both the same for Post and Site editor since they are using the same shared components where I made the changes.

Chrome performance profiler (no throttling)

| Before (~200ms) | After (~66ms) |
| - | - |
| ![image](https://user-images.githubusercontent.com/2256104/96706225-9887f080-1396-11eb-9ab3-005910df2f94.png) | ![image](https://user-images.githubusercontent.com/2256104/96706283-accbed80-1396-11eb-8520-df26179a986d.png) |

React performance profiler (no throttling)

| Before (~70ms) | After (~2ms) |
| - | - |
| ![image](https://user-images.githubusercontent.com/2256104/96706775-501d0280-1397-11eb-8f7b-813b4557d506.png) | ![image](https://user-images.githubusercontent.com/2256104/96706703-37ace800-1397-11eb-970f-9793265c51fc.png) |

Highlighting rerendered components

| Before | After |
| - | - |
| ![ezgif com-gif-maker (6)](https://user-images.githubusercontent.com/2256104/96707522-447e0b80-1398-11eb-9703-cc57aaf075e2.gif) | ![ezgif com-gif-maker (5)](https://user-images.githubusercontent.com/2256104/96707496-3b8d3a00-1398-11eb-9b9d-82675571ec37.gif) |


## Types of changes
Performance

## Checklist:
